### ISSUE TITLE
Type hints mask E251 whitespace-around-equals errors

### ISSUE THREAD

**Issue by nchammas** [PyCQA/pycodestyle#675]
Repro:

```sh
$ pycodestyle <(
cat << EOM
def le_func(a: int = 4):
    pass
EOM
)
```

No errors for that, even though there should be.

However, without the type hint:

```sh
$ pycodestyle <(
cat << EOM
def le_func(a = 4):
    pass
EOM
)
/dev/fd/63:1:14: E251 unexpected spaces around keyword / parameter equals
/dev/fd/63:1:16: E251 unexpected spaces around keyword / parameter equals
```

What I'm running:

```sh
$ python --version
Python 3.5.2
$ pycodestyle --version
2.3.1
```

**Comment by sigmavirus24** [comment]
This is intentional, per PEP-0008's advice.

See also:

- https://github.com/PyCQA/pycodestyle/issues/551
- https://github.com/PyCQA/pycodestyle/issues/507
- https://github.com/PyCQA/pycodestyle/issues/496

**Comment by nchammas** [comment]
I see. I searched on here for ["type hint"](https://github.com/PyCQA/pycodestyle/issues?q=is%3Aissue+type+hint+is%3Aclosed) and missed those issues. Sorry about the false alarm.
